### PR TITLE
POC how an emulated async strategy may look like

### PR DESCRIPTION
**Pull request by Nyholm**
This PR should not be merged. I just want to show how I, as an application author, may force discovery to emulate an async client. 

Ref: https://github.com/php-http/discovery/issues/57

### CODE REVIEW

**Comment by sagikazarmark** [comment]
I don't really like this. It adds an optional dependency. And IMO this is not how resources should be resolved.

Also, this is not much code: I would just have something similar in the docs or cookbook.

**Comment by Nyholm** [comment]
Sorry if I wasn't clear. The EmulateAsyncClientStrategy does not belong in this repo for the reason you mention. But if we should allow a class similar to EmulateAsyncClientStrategy in the cookbook we have to make the changes in the ClassDiscovery and the the other *Discovery classes. 

I just want your feedback/opinions/suggestions for the changes in the ClassDiscovery and the *Discovery classes.

File: src/ClassDiscovery.php [hunk 1]
```diff
@@ -36,7 +36,7 @@
      *
      * @param string $type
      *
-     * @return string
+     * @return string|\Closure
```
Comment (Nyholm) [inline src/ClassDiscovery.php]: Do we want to allow this function to return a Closure?

**Comment by Nyholm** [comment]
Im closing this PR in favor for #66.

**Pull request closed without merging**